flake.nix

```diff
@@ -0,0 +1,30 @@
+{
```

Comment:
Does the modus operandi of Nix really require injecting these files into every single repository it manages?

Reply:
The reason of adding nix here is simply add dependencies to repositories, providing a reproducible and cross platform/os experience for users. For test dependencies, they can be simply cached and speeding up the CI flow.

Reply:
I understand the purpose of Nix, I just find it surprising that making packages Nix-compatible seems to require injecting a non-trivial amount of boilerplate into each repo. This is fine then.

Reply:
I know it's suffers, since Chisel are depending more and more and more native dependencies like CIRCT, espresso, slang... It's hard to distributing them for different users. Thus I eventually choose nix(although I hate it).

Reply:
Understood,